h5py/_conv.pyx

```diff
@@ -762,22 +762,22 @@ cdef int conv_ndarray2vlen(void* ipt,
         vlen_t* in_vlen = <vlen_t*>opt
         void* data
         cnp.ndarray ndarray
-        size_t len
+        size_t len, nbytes
         PyObject* buf_obj0
 
     buf_obj0 = buf_obj[0]
     ndarray = <cnp.ndarray> buf_obj0
     len = ndarray.shape[0]
+    nbytes = len * max(outtype.get_size(), intype.get_size())
 
-    if outtype.get_size() > intype.get_size():
-        data = emalloc(outtype.get_size() * len)
-    else:
-        data = emalloc(intype.get_size() * len)
-    memcpy(data, ndarray.data, intype.get_size() * len)
-    H5Tconvert(intype.id, outtype.id, len, data, NULL, H5P_DEFAULT)
+    #with nogil:
+    if 1:
```

Comment:
Leftover?

Reply:
indeed. 
I was not sure to make the call to emalloc gil-free or not. It is a subject which is worth discussing I believe. What is your opinion ?
Shall we focus on making the  actual IO gil-free ?

Reply:
Do you know how big a chunk of data HDF5 prefers to pass to converters? I guess that the bigger the chunks are, the more it makes sense to release the GIL here.

Reply:
Vlen are often used to store strings of comments ... so the answer is probably "not much".